pkg/ccl/backupccl/backup_planning.go

```diff
@@ -255,43 +238,18 @@ func getLogicallyMergedTableSpans(
 		// interleaved contains the tableID/indexID of the furthest ancestor in
 		// the interleaved chain.
 		if lhsIndex.IsInterleaved() || rhsIndex.IsInterleaved() {
+			mergedSpan.EndKey = lhsSpan.EndKey
 			mergedIndexSpans = append(mergedIndexSpans, mergedSpan)
 			mergedSpan = rhsSpan
-		} else {
-			var foundDroppedKV bool
-			// Iterate over all index IDs between the two candidates (lhs and
-			// rhs) which may be logically merged. These index IDs represent
-			// non-public (and perhaps dropped) indexes between the two public
-			// index spans.
-			for i := lhsIndexID + 1; i < rhsIndexID; i++ {
-				// If we find an index which has been dropped but not gc'ed, we
-				// cannot merge the lhs and rhs spans.
-				foundDroppedKV, err = checkForKVInBounds(lhsSpan.EndKey, rhsSpan.Key, endTime)
-				if err != nil {
-					// If we're unable to check for KVs in bounds, assume that we've found
-					// one. It's always safe to assume that since we won't merge over this
-					// span. One possible error is a GC threshold error if this schema
-					// revision is older than the configured GC window on the span we're
-					// checking.
-					log.Warningf(ctx, "error while scanning [%s, %s) @ %v: %v",
-						lhsSpan.EndKey, rhsSpan.Key, endTime, err)
-					foundDroppedKV = true
-				}
-				// If we find an index that is being added, don't merge the spans. We
-				// don't want to backup data that is being backfilled until the backfill
-				// is complete. Even if the backfill has not started yet and there is no
-				// data we should not back up this span since we want these spans to
-				// appear as introduced when the index becomes PUBLIC.
-				// The indexes will appear in introduced spans because indexes
-				// will never go from PUBLIC to ADDING.
-				_, foundAddingIndex := addingIndexIDs[i]
-				if foundDroppedKV || foundAddingIndex {
-					mergedSpan.EndKey = lhsSpan.EndKey
-					mergedIndexSpans = append(mergedIndexSpans, mergedSpan)
-					mergedSpan = rhsSpan
-					break
-				}
-			}
+		} else if lhsIndexID+1 < rhsIndexID {
+			// If there are non-public indexes between the lhs and rhs public indexes
+			// we do not attempt to merge the spans. These non-public indexes could
+			// be in an adding or dropped state.
+			mergedSpan.EndKey = lhsSpan.EndKey
+			mergedIndexSpans = append(mergedIndexSpans, mergedSpan)
+			mergedSpan = rhsSpan
+		} else if lhsIndexID+1 > rhsIndexID {
```

Comment:
[nit] perhaps move this assertion right up after the IDs are created and test on `==`

Reply:
done.